index.html

```diff
@@ -174,95 +180,122 @@
       </div>
       <section>
         <h2>
-          Syntax
+          Validation
         </h2>
         <p>
-          When URLs are used for payment method identifiers they MUST be
-          absolute URLs including, at most, a <a>scheme</a>, <a>host</a>,
-          <a>port</a> and <a>path</a>. The URL must be a <a>potentially
-          trustworthy URL</a> as defined in the [[SECURE-CONTEXTS]]
-          specification. Identifier URLs MUST have null <a>query</a> and
-          <a>fragment</a> values.
+          The steps to <dfn data-lt="validate|valid">validate a URL-based
+          payment method identifier</dfn> are given by the following algorithm.
+          The algorithm takes a <var>input</var> string as input and returns
+          either true or an error:
         </p>
+        <ol class="algorithm">
```

Comment:
OK, after removing the matching algorithm, I see your perspective. But, I think that it's still a better algorithm if it's more "strongly typed", i.e. operates on URLs.

I have a separate issue which is that I think it should return true or false, instead of returning true or throwing a TypeError. E.g. in the payment-method-manifest spec we aren't in a context where throwing makes sense, so we'd have to "catch" the exception, even though we're not even in a JavaScript context where exceptions are sensible to talk about.